docs/documentation/helm.md

```diff
@@ -0,0 +1,276 @@
+  <artifactId>helm-annotations</artifactId>
+  <version>{{site.data.project.release.current-version}}</version>
+</dependency>
+```
+
+This dependency will generate the following Helm resources:
+- Chart.yaml
+- values.yaml
+- templates/*.yml the generated resources by Dekorate
+- <chart name>-<chart version>-helm.tar.gz
+
+#### Getting Started
+
+To generate the Helm resources, you first need to configure the Helm chart via properties:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+```
+
+Or annotate one of your Java source files with the [@HelmChart](annotations/helm-annotations/src/main/java/io/dekorate/helm/annotation/HelmChart.java) annotation:
+
+```java
+@HelmChart(name = "myChart", description = "Description of my Chart")
+@SpringBootApplication
+public class Main {
+
+  public static void main(String[] args) {
+    SpringApplication.run(Main.class, args);
+  }
+}
+```
+
+Once, you have configured the Helm chart, you can generate the Helm resources under the folder `target/classes/META-INF/dekorate/helm/<chart name>/` with:
+
+    mvn clean package
+
+Assuming you're using the [Kubernetes]({{site.baseurl}}/docs/kubernetes) Dekorate extension, the Helm resources will include by default the following templates at `target/classes/META-INF/dekorate/helm/<chart name>/templates/`:
+- deployment.yaml
+- ingress.yaml
+- service.yaml
+- NOTES.txt
+
+**Note**: the Helm annotation is also compatible with the [Openshift]({{site.baseurl}}/docs/openshift) Dekorate extension.
+
+#### Mapping Values
+
+By default, Dekorate will generate the Helm values file (`values.yaml`) by mapping the following pre-configured properties:
+
+- The Kubernetes/Openshift replicas
+- The Kubernetes/Openshift image
+- The Kubernetes/Openshift Env Var values (only for plain values - secrets or configmaps are not supported yet)
+- The Kubernetes ingress host
+- The Openshift S2i builder image
+
+For example, if you set 3 replicas for your deployment:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+
+# Set replicas to 3
+dekorate.kubernetes.replicas=3
+```
+
+Dekorate will generate the next Helm values file at `target/classes/META-INF/dekorate/helm/<chart name>/values.yaml`:
+
+```yaml
+---
+myModule:
+  replicas: 3
+```
+
+**Note**: `myModule` is the name of your project.
+
+And the Deployment file at `target/classes/META-INF/dekorate/helm/<chart name>/templates/deployment.yaml` will have a reference to this value:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myModule
+spec:
+  replicas: {{ .Values.myModule.replicas }}
+```
+
+This is done transparently to users.
+
+##### Mapping user properties using JSONPath expressions
+
+Let's see the following YAML resource:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: helm-on-kubernetes-example
+...
+```
+
+Dekorate will not replace the property at `metadata.name` with `{{ .Values.myModule.name }}`.
```

Comment:
There is still something which is missing as you dont do a transition within the text between "Let's see the following YAML resource:" AND "Dekorate will not replace the property at `metadata.name` with `{{ .Values.myModule.name }}`.".

Please add a line doing such a transition as we still dont know yet what you would like to do

Reply:
Done